src/mpid/ch4/generic/mpidig_globals.c

```diff
@@ -11,6 +11,49 @@
 
 #include "mpidimpl.h"
 #include "mpidig.h"
+#include "ch4_impl.h"
 
 MPIDIG_global_t MPIDIG_global = { {0}
 };
+
+#undef FUNCNAME
+#define FUNCNAME MPIDIG_comm_abort
+#undef FCNAME
+#define FCNAME MPL_QUOTE(FUNCNAME)
+void MPIDIG_comm_abort(MPIR_Comm * comm, int exit_code)
+{
+    int mpi_errno = MPI_SUCCESS;
+    int dest;
+    int buf = exit_code;
+    MPIR_Request *sreq = NULL;
+    MPIDI_CH4U_hdr_t am_hdr;
+
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDIG_COMM_ABORT);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDIG_COMM_ABORT);
+
+    am_hdr.src_rank = comm->rank;
+    am_hdr.tag = exit_code;
+    am_hdr.context_id = comm->context_id + MPIR_CONTEXT_INTRA_PT2PT;
+
+    for (dest = 0; dest < comm->size; dest++) {
```

Comment:
This won't compile, though it gets changed later so it's not noticeable. It should be either `comm->local_size` or `MPIR_Comm_size(comm)`.

Reply:
Maybe we should just squash this with the next commit that gets rid of this line. I don't know if there's benefit to having separate commits for the intra/intercomm versions. And just to reiterate, this function probably needs to go back to returning an `int` for the intercomm version.